File: py/src/practise150/p18_permutation_in_string__test.py
from src.practise150.p18_permuation_in_string import Solution

def test_permutation_in_string():
    test_cases=[
        ("abc", "lecabee", True),
        ("abc", "lecaabee", False)
    ]
    s = Solution()
    for test_case in test_cases:
        assert test_case[2] == s.check_permutation(test_case[0], test_case[1])

File: go/practise150/p18_permutation_in_string_test.go
package practise150

import (
	"fmt"
	"testing"
)

func TestPermutationInString(t *testing.T) {
	test_cases := []struct {
		input1   string
		input2   string
		expected bool
	}{
		{
			input1:   "abc",
			input2:   "lecabee",
			expected: true,
		},
		{
			input1:   "abc",
			input2:   "lecaabee",
			expected: false,
		},
	}

	for idx, tc := range test_cases {
		t.Run(fmt.Sprintf("Test_case_no:%d", idx+1), func(t *testing.T) {
			actual := CheckPermutation(tc.input1, tc.input2)
			if actual != tc.expected {
				t.Errorf("Testing permutation in string. Input1: %s, Input2: %s, Expected: %v, Actual: %v.", tc.input1, tc.input2, tc.expected, actual)
			}
		})
	}
}
